java-api-core/src/com/kinvey/java/core/MediaHttpDownloader.java

```diff
@@ -270,7 +270,7 @@ public FileMetaData download(FileMetaData metaData, OutputStream out) throws IOE
             }
 
             // set Range header (if necessary)
-            if (bytesDownloaded != 0 || currentRequestLastBytePos != -1) {
+            if (metaData.getSize() != 0 && (bytesDownloaded != 0 || currentRequestLastBytePos != -1)) {
```

Comment:
@yuliya-guseva If the file size is zero, wouldn't the `bytesDownloaded` also be zero?